src/map.cpp

```diff
@@ -3100,37 +3071,47 @@ bool map::has_nearby_fire( const tripoint &p, int radius )
     return false;
 }
 
-void map::mop_spills( const tripoint &p ) {
+bool map::mop_spills( const tripoint &p )
+{
+    bool retval = false;
     auto items = i_at( p );
     for( auto it = items.begin(); it != items.end(); ) {
-        if( it->made_of(LIQUID) ) {
+        if( it->made_of( LIQUID ) ) {
             it = items.erase( it );
+            retval = true;
         } else {
             it++;
         }
     }
-    remove_field( p, fd_blood );
-    remove_field( p, fd_blood_veggy );
-    remove_field( p, fd_blood_insect );
-    remove_field( p, fd_blood_invertebrate );
-    remove_field( p, fd_gibs_flesh );
-    remove_field( p, fd_gibs_veggy );
-    remove_field( p, fd_gibs_insect );
-    remove_field( p, fd_gibs_invertebrate );
-    remove_field( p, fd_bile );
-    remove_field( p, fd_slime );
-    remove_field( p, fd_sludge );
+
+    const field &fld = field_at( p );
+    std::vector<field_id> to_check = {fd_blood, fd_blood_veggy,
+                                      fd_blood_insect, fd_blood_invertebrate, fd_gibs_flesh, fd_gibs_veggy,
+                                      fd_gibs_insect, fd_gibs_invertebrate, fd_bile, fd_slime, fd_sludge
+                                     };
+    for( unsigned i = 0; i < to_check.size(); i++ ) {
+        if( fld.findField( to_check[i] ) != 0 ) {
```

Comment:
Thanks, I'll remember that next time, but this time I'll remove that function call per codemime's comment.